src/Registry.php

```diff
@@ -827,4 +840,104 @@ protected function toFlatten($separator = null, $data = null, array &$array = []
             $array[$key] = $v;
         }
     }
+
+    /**
+     * Magic method to access separator property.
+     *
+     * @param  string  $name  The name of the property.
+     *
+     * @return string|null A value if the property name is valid, null otherwise.
+     *
+     * @since       __DEPLOY_VERSION__
+     * @deprecated  3.0  This is a B/C proxy for deprecated read accesses
+     */
+    public function __get($name)
+    {
+        switch ($name) {
+            case 'separator':
+                \trigger_deprecation(
+                    'joomla/registry',
+                    '__DEPLOY_VERSION__',
+                    'The $separator parameter will be removed in version 3.',
+                    self::class,
+                    self::class
+                );
+
+                return $this->separator;
+
+            default:
+                if (property_exists($this, $name)) {
+                    throw new \RuntimeException(
+                        \sprintf(
+                            'Cannot access protected or private property %s::$%s',
+                            __CLASS__,
+                            $name
+                        )
+                    );
+                }
```

Comment:
If I'm not completely wrong, this would break child classes with implement own public properties.
We need to check the reflection if the property is private or protected.

Reply:
Nope, check my note before: The magic methods does not called for existing public/accesible properties.
That why we have this https://github.com/joomla/joomla-cms/blob/1995c41f6f2075f6916b337777888a68de9fefb9/libraries/src/Form/FormField.php#L666-L668 in CMS.

Reply:
right then it works

Reply:
PHP 7.4 and bad design... but this is what you get.
```
class small
{
	private $name;
	protected $key;
	public $value;

	public function __construct()
	{
		$this->dynamic = 'wrong';
	}

	public function testMe($name)
	{
		if (property_exists($this, $name)) {
			return "FOUND: $name<br />";
		}
		return "IS NOT FOUND: $name<br />";
	}
}

class medium extends small
{
	private $foo;
	protected $bar;
	public $duck;

	public function __construct()
	{
		$this->dry = 'ice';
		parent::__construct();
	}
}

// just inherited again
$andYetAgain =  new medium();

echo '<h3>Medium</h3>';
// few tests
$andYetAgain->chipping = 'all in one go';
echo $andYetAgain->testMe('chipping');  // FOUND: chipping
echo $andYetAgain->testMe('name');      // FOUND: name
echo $andYetAgain->testMe('key');       // FOUND: key
echo $andYetAgain->testMe('value');     // FOUND: value
echo $andYetAgain->testMe('dynamic');   // FOUND: dynamic
echo $andYetAgain->testMe('foo');       // FOUND: foo
echo $andYetAgain->testMe('bar');       // FOUND: bar
echo $andYetAgain->testMe('duck');      // FOUND: duck
echo $andYetAgain->testMe('dry');       // FOUND: dry
echo $andYetAgain->testMe('notMe');     // IS NOT FOUND: notMe

class large extends medium
{
	private $dog;

	public function __construct()
	{
		$this->cat = 'hmmm';
		parent::__construct();
	}
}

// just inherited again
$andYetAgain =  new large();
echo '<h3>Large</h3>';
// few tests
$andYetAgain->chipping = 'all in one go';
echo $andYetAgain->testMe('chipping');  // FOUND: chipping
echo $andYetAgain->testMe('name');      // FOUND: name
echo $andYetAgain->testMe('key');       // FOUND: key
echo $andYetAgain->testMe('value');     // FOUND: value
echo $andYetAgain->testMe('dynamic');   // FOUND: dynamic
echo $andYetAgain->testMe('foo');       // IS NOT FOUND: foo
echo $andYetAgain->testMe('bar');       // FOUND: bar
echo $andYetAgain->testMe('duck');      // FOUND: duck
echo $andYetAgain->testMe('dry');       // FOUND: dry
echo $andYetAgain->testMe('dog');       // FOUND: dog
echo $andYetAgain->testMe('cat');       // FOUND: cat
echo $andYetAgain->testMe('notMe');     // IS NOT FOUND: notMe
```

Reply:
Same result in PHP 8.0

Reply:
hmm don't mind me... I was looking at the wrong issue.

Reply:
https://3v4l.org/s3EP7

Reply:
https://3v4l.org/fgOsF (this is a better illustration of the actual issue)

Reply:
That is correct, a public properties will work as they were before and for a dynamic properties the deprecation will be trigered only once.